src/main/java/frc/robot/Robot.java

```diff
@@ -44,15 +52,28 @@ public void robotPeriodic() {
     // and running subsystem periodic() methods.  This must be called from the robot's periodic
     // block in order for anything in the Command-based framework to work.
 
+    Threads.setCurrentThreadPriority(true, 99);
+
     EagleUtil.clearCachedPose();
+    double startTime = HALUtil.getFPGATime();
 
     CommandScheduler.getInstance().run();
 
+    DogLog.log("Loop Time/Command Scheduler", (HALUtil.getFPGATime() - startTime) / 1000);
+
+    double endTime = HALUtil.getFPGATime();
+
     m_robotContainer.periodic();
 
+    DogLog.log("Loop Time/Robot Container", (HALUtil.getFPGATime() - endTime) / 1000);
+
+    gcStatsCollector.update();
+
     double currentTime = HALUtil.getFPGATime();
-    DogLog.log("Loop Time", (currentTime - prevTime) / 1000);
+    DogLog.log("Loop Time/Total", (currentTime - prevTime) / 1000);
     prevTime = currentTime;
+
+    Threads.setCurrentThreadPriority(false, 10);
```

Comment:
We should save the initial thread state (boolean realtime and priority) at the start of the function, and restore those initial values on the way out.

I would also use a `finally` block to restore it all.